File: frontend/types/social_publishing.ts
/**
 * Social Publishing Types
 * ======================
 * TypeScript types for social media publishing workflow
 */

export type Channel = 'facebook' | 'instagram' | 'website';
export type DraftStatus = 'draft' | 'generated' | 'edited' | 'ready' | 'publishing' | 'published' | 'failed';

export interface AIDraft {
    id?: string;
    propertyId: string;
    language: string;       // e.g. 'en-IN', 'mr-IN'
    channel: Channel;
    title: string;
    body: string;
    hashtags: string[];
    mediaIds: string[];
    contactIncluded: boolean;
    status: DraftStatus;
    editedBy?: string;
    updatedAt?: string;
    createdAt?: string;
}

export interface SocialAccount {
    id?: string;
    agentId: string;
    platform: Channel;
    pageId?: string;
    isActive: boolean;
    createdAt?: string;
    updatedAt?: string;
}

export interface SocialPost {
    id?: string;
    draftId: string;
    platformPostId?: string;
    platform: Channel;
    status: 'published' | 'failed' | 'scheduled';
    errorMessage?: string;
    publishedAt?: string;
    createdAt?: string;
}

export interface GenerateContentRequest {
    propertyId: string;
    language: string;
    channels: Channel[];
    tone?: string;
    length?: string;
    agentId: string;
}

export interface GenerateContentResponse {
    drafts: AIDraft[];
}

export interface UpdateDraftRequest {
    title?: string;
    body?: string;
    hashtags?: string[];
    contactIncluded?: boolean;
    mediaIds?: string[];
    status?: DraftStatus;
}

export interface MarkReadyRequest {
    draftIds: string[];
}

export interface PublishRequest {
    draftIds: string[];
    scheduleAt?: string;
}

export interface PublishResponse {
    jobId: string;
    message: string;
}

export interface DraftsResponse {
    propertyId: string;
    language: string;
    drafts: AIDraft[];
}

export interface ContactInfo {
    name: string;
    phone: string;
    whatsapp?: string;
    email?: string;
    website?: string;
}

export interface PropertyContext {
    id: string;
    title: string;
    description: string;
    price: number;
    location: string;
    propertyType: string;
    bedrooms: number;
    bathrooms: number;
    areaSqft: number;
    amenities: string[];
    features: string[];
    images: string[];
}

export interface AIGenerationContext {
    property: PropertyContext;
    agent: ContactInfo;
    language: string;
    channel: Channel;
    tone: string;
    length: string;
}

// UI State Types
export interface PublishingUIState {
    selectedProperty: PropertyContext | null;
    selectedLanguages: string[];
    selectedChannels: Channel[];
    drafts: Map<string, Map<Channel, AIDraft>>; // language -> channel -> draft
    loadingStates: Map<string, boolean>; // language -> loading
    currentLanguage: string;
    currentChannel: Channel;
}

export interface CharLimits {
    maxChars: number;
    maxHashtags: number;
    warningThreshold: number;
}

export interface ToneOptions {
    friendly: string;
    luxury: string;
    investor: string;
}

export interface LengthOptions {
    short: string;
    medium: string;
    long: string;
}

// Component Props
export interface LanguageSelectorProps {
    selectedLanguages: string[];
    onLanguageChange: (languages: string[]) => void;
    onLanguageAdded: (language: string) => void;
    loadingStates: Map<string, boolean>;
}

export interface ChannelSelectorProps {
    selectedChannels: Channel[];
    onChannelChange: (channels: Channel[]) => void;
}

export interface AIContentPanelProps {
    property: PropertyContext;
    language: string;
    channel: Channel;
    draft: AIDraft | null;
    onDraftUpdate: (draft: AIDraft) => void;
    onRegenerate: () => void;
    onImprove: (type: string) => void;
    onMarkReady: () => void;
    isLoading: boolean;
}

export interface ContentEditorProps {
    draft: AIDraft;
    onUpdate: (updates: Partial<AIDraft>) => void;
    limits: CharLimits;
}

export interface PreviewPanelProps {
    draft: AIDraft;
    channel: Channel;
}

export interface PublishBarProps {
    readyDrafts: AIDraft[];
    onPublish: () => void;
    onSchedule: (scheduleAt: string) => void;
    onExport: () => void;
    isPublishing: boolean;
}

File: backend/app/schemas/social_publishing.py
"""
Social Publishing Schemas
========================
Pydantic schemas for social media publishing workflow
"""

from typing import List, Optional, Dict, Any, Literal
from pydantic import BaseModel, Field
from datetime import datetime
from enum import Enum

class Channel(str, Enum):
    FACEBOOK = "facebook"
    INSTAGRAM = "instagram"
    WEBSITE = "website"

class DraftStatus(str, Enum):
    DRAFT = "draft"
    GENERATED = "generated"
    EDITED = "edited"
    READY = "ready"
    PUBLISHING = "publishing"
    PUBLISHED = "published"
    FAILED = "failed"

class SocialAccount(BaseModel):
    """Social media account configuration"""
    id: Optional[str] = None
    agent_id: str
    platform: Channel
    page_id: Optional[str] = None  # Facebook Page ID or Instagram Business Account ID
    access_token: str  # Server-only, encrypted
    is_active: bool = True
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

class AIDraft(BaseModel):
    """AI-generated content draft"""
    id: Optional[str] = None
    property_id: str
    language: str  # e.g., 'en-IN', 'mr-IN'
    channel: Channel
    title: str
    body: str
    hashtags: List[str] = Field(default_factory=list)
    media_ids: List[str] = Field(default_factory=list)
    contact_included: bool = True
    status: DraftStatus = DraftStatus.DRAFT
    edited_by: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

class SocialPost(BaseModel):
    """Published social media post"""
    id: Optional[str] = None
    draft_id: str
    platform_post_id: Optional[str] = None  # Facebook/Instagram post ID
    platform: Channel
    status: Literal["published", "failed", "scheduled"] = "published"
    error_message: Optional[str] = None
    published_at: Optional[datetime] = None
    created_at: Optional[datetime] = None

class GenerateContentRequest(BaseModel):
    """Request to generate AI content"""
    property_id: str
    language: str
    channels: List[Channel]
    tone: Optional[str] = "friendly"
    length: Optional[str] = "medium"
    agent_id: str

class GenerateContentResponse(BaseModel):
    """Response from content generation"""
    drafts: List[Dict[str, Any]]

class UpdateDraftRequest(BaseModel):
    """Request to update a draft"""
    title: Optional[str] = None
    body: Optional[str] = None
    hashtags: Optional[List[str]] = None
    contact_included: Optional[bool] = None
    media_ids: Optional[List[str]] = None
    status: Optional[DraftStatus] = None

class MarkReadyRequest(BaseModel):
    """Request to mark drafts as ready"""
    draft_ids: List[str]

class PublishRequest(BaseModel):
    """Request to publish drafts"""
    draft_ids: List[str]
    schedule_at: Optional[datetime] = None

class PublishResponse(BaseModel):
    """Response from publishing"""
    job_id: str
    message: str

class DraftsResponse(BaseModel):
    """Response with drafts for a property"""
    property_id: str
    language: str
    drafts: List[AIDraft]

class ContactInfo(BaseModel):
    """Agent contact information for content"""
    name: str
    phone: str
    whatsapp: Optional[str] = None
    email: Optional[str] = None
    website: Optional[str] = None

class PropertyContext(BaseModel):
    """Property context for AI generation"""
    id: str
    title: str
    description: str
    price: float
    location: str
    property_type: str
    bedrooms: int
    bathrooms: int
    area_sqft: int
    amenities: List[str]
    features: List[str]
    images: List[str]

class AIGenerationContext(BaseModel):
    """Complete context for AI generation"""
    property: PropertyContext
    agent: ContactInfo
    language: str
    channel: Channel
    tone: str = "friendly"
    length: str = "medium"
